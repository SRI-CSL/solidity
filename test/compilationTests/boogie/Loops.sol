pragma solidity ^0.4.17;

contract Loops {

    function whileLoopFunc(uint param) pure public returns (uint) {
        uint i = 0;
        uint result = param;

        while (i < 10) {
            result = result + 1;
            i = i + 1;
        }

        return result;
    }

    function forLoopFunc(uint param) pure public returns (uint) {
        uint result = param;

        for (uint i = 0; i < 10; i = i + 1) {
            result = result + 1;
        }

        return result;
    }

    function breakLoop(uint param) pure public returns (uint) {
        uint i = 0;
        uint result = param;

        while (i < 10) {
            result = result + 1;
            if (result > 100) break;
            i = i + 1;
        }

        return result;
    }

}