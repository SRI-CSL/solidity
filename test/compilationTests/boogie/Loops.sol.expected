Verifying Loops::__verifier_main: OK
No errors found.
